### PR TITLE
[fix] corda-enterprise flow and logic

### PR DESCRIPTION
**Pull request by sownak**
Signed-off-by: sownak.roy <sownak.roy@accenture.com>

**Changelog**
- Remove unused chart pki-node
- Update create/certificates/ambassador to make it more generic so that less code can be used.
- Update deploy-network playbook to make it more relevant for usage

**Next Steps**
- Notary deployment is not working most probably because PVC is not attached for the notary
- Node deployment has not been tested, but needs to follow similar pattern as notary.

### CODE REVIEW

**Comment by sownak** [comment]
Notary fixed with following changes:
- Ambassador certs are changed so that unnecessary certificate chaining is not needed. Ambassador tls certs should be independent of the core certificate chain
- idman.crt and networkmap.crt import should be done with correct alias
- sslkeystore should be passed when starting notary as the ketstore is reset when using h2 db 
- networkmap should have idman.crt imported using keytool (hence added the dockerfile to run networkmap as root

Notary is up and running as can be verified by these URLs:
- curl -k https://<nmap public uri>:8443/network-map-user/node-infos
- curl -k https://<nmap public uri>:8443/network-map-user/network-map

File: platforms/corda-ent/charts/notary/templates/deployment.yaml [hunk 1]
```diff
@@ -217,26 +212,40 @@ spec:
...
           chmod +x bin/*
+
+          # to clean network-parameters on every restart
+          rm -rf ${BASE_DIR}/network-parameters
+
+          # add idman and networkmap certificates to java keystore
+          yes | keytool -importcert -file ${BASE_DIR}/certificates/networkmap.crt -storepass changeit -alias {{ .Values.networkServices.networkMapDomain }} -keystore /usr/lib/jvm/zulu-8-amd64/jre/lib/security/cacerts
+          yes | keytool -importcert -file ${BASE_DIR}/certificates/idman.crt -storepass changeit -alias {{ .Values.networkServices.idmanDomain }} -keystore /usr/lib/jvm/zulu-8-amd64/jre/lib/security/cacerts          
           bin/run.sh
+          tail -f /dev/null
```
Comment (suvajit-sarkar) [inline platforms/corda-ent/charts/notary/templates/deployment.yaml:230]: can be removed

File: platforms/corda-ent/images/networkmap.dockerfile [hunk 2]
```diff
@@ -0,0 +1,4 @@
+FROM corda/enterprise-networkmap:1.2-zulu-openjdk8u242
+
+ENV ACCEPT_LICENSE=Y
+USER root
```
Comment (suvajit-sarkar) [inline platforms/corda-ent/images/networkmap.dockerfile:4]: newline please

**Comment by suvajit-sarkar** [comment]
This PR also resolves the issue raised in #935

**Review by jagpreetsinghsasan — APPROVED** [review]

**Review by sauveergoel — APPROVED** [review]

File: platforms/corda-ent/charts/notary/templates/pvc.yaml [hunk 3]
```diff
@@ -19,4 +19,4 @@ spec:
   storageClassName: {{ .Values.storage.name }}
   resources:
     requests:
-      storage: 64Mi
+      storage: 512Mi
```
Comment (abevers) [inline platforms/corda-ent/charts/notary/templates/pvc.yaml:22]: Storage size is hardcoded. Move this to `values.yaml` and add to `notary.tpl` file.

File: platforms/corda-ent/charts/notary/values.yaml [hunk 4]
```diff
@@ -85,8 +85,12 @@ rpcSettingsUseSsl: false
 networkServices:
   # Ex. doormanURL: "http://idman-internal.test-kd:10000"
   doormanURL:
+  idmanName:
```
Comment (abevers) [inline platforms/corda-ent/charts/notary/values.yaml:88]: Could structure this as such, both are fine
```
  networkServices:
    idman:
      url:
      name: 
      domain:   
    networkmap:
      url:
      name: 
      domain:
```

Comment (sownak) [reply platforms/corda-ent/charts/notary/values.yaml:88]: These are to be fixed by whoever is doing the notary refactoring

**Pull request merged**